### PR TITLE
add Java 17 and 21 to nightly.yml

### PR DESCRIPTION
**Pull request by pjfanning**
delaying while #261 is looked at

### CODE REVIEW

**Review by raboof — COMMENTED** [review]
I wonder if we're over-doing it with the tested Java versions - perhaps 'minimum supported' and 'latest LTS supported' should be enough?

Change LGTM, one comment on the trigger.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ name: Nightly Tests
 
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (pjfanning) [inline .github/workflows/nightly.yml:5]: added temporarily just to have this change tested

**Comment by pjfanning** [comment]
Mockito issue in Java 17 and 21. Looks like we might need to add `--add-opens` settings.

```
[error] Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @23828349
[error] 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
[error] 	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
[error] 	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
[error] 	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
[error] 	at org.mockito.cglib.core.ReflectUtils$2.run(ReflectUtils.java:57)
```

**Comment by pjfanning** [comment]
after the mockito upgrade, the tests now pass in java 17 and 21

**Review by nvollmar — APPROVED** [review]
lgtm

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -12,13 +12,9 @@ jobs:
...
       fail-fast: false
       matrix:
-        include:
-          - { javaVersion: 8,  scalaVersion: 2.12, pekkoVersion: 'default' }
-          - { javaVersion: 8,  scalaVersion: 2.12, pekkoVersion: '1.0.x' }
-          - { javaVersion: 8,  scalaVersion: 2.13, pekkoVersion: 'default' }
-          - { javaVersion: 8,  scalaVersion: 2.13, pekkoVersion: '1.0.x' }
-          - { javaVersion: 8,  scalaVersion: 3.3,  pekkoVersion: 'default' }
-          - { javaVersion: 8,  scalaVersion: 3.3,  pekkoVersion: '1.0.x' }
+        scalaVersion: ['2.12', '2.13', '3.3']
```
Comment (Roiocam) [inline .github/workflows/nightly.yml:15]: cool

**Pull request merged**